docs/usage.rst

```diff
@@ -85,6 +83,49 @@ The ``namelist`` item is special in that it is explicitly stored in the ``config
 fv3gfs model, individual namelists are specified for various components of the model. As an example, the
 vertical resolution can be accessed via ``config['namelist']['fv_core_nml']['npz']``.
 
+Running the model with fv3run
+-----------------------------
+
+`fv3config` provides a tool for running the python-wrapped model called `fv3run`.
+
+.. code-block:: bash
+
+    $ fv3run --help
+    usage: fv3run [-h] [--runfile RUNFILE] [--dockerimage DOCKERIMAGE]
+                  [--keyfile KEYFILE]
+                  config outdir
+
+    Run the FV3GFS model. Will use google cloud storage key at
+    $GOOGLE_APPLICATION_CREDENTIALS by default.
+
+    positional arguments:
+      config                location of fv3config yaml file
+      outdir                location to copy final run directory
+
+    optional arguments:
+      -h, --help            show this help message and exit
+      --runfile RUNFILE     location of python script to execute with mpirun
+      --dockerimage DOCKERIMAGE
+                            if passed, execute inside a docker image with the
+                            given name
+      --keyfile KEYFILE     google cloud storage key to use for cloud copy
+                            commands
+
+The only required inputs are ``config``, which specifies a yaml file containing the
```

Comment:
These docs are great. It would be nice to have a working end-to-end example and perhaps a default docker image so that people don't need to specify that manually. Is there a GCR image for fv3gfs-python?

Reply:
Thanks! It's basically designed to work with the fv3gfs-python docker image, so perhaps that could be default... but if users have to specify an option anyways to enable docker then I figured why not do it by having them say what image they want to use? fv3gfs-python should be on GCR.

I was told that if you try to docker run an image that has a remote url it will fetch from the remote if you don't have it locally, but in practice I've found the docker run command fails whenever that happens saying you don't have the image locally. Am I doing something wrong?

Reply:
It should download. I think it could be because the image doesn't exist in GCR yet.

Reply:
I presume this fixed itself when you asked me to manually push an image to GCR?

Reply:
I can know use the image, but still cannot execute an end-to-end workflow:
```
In [1]: import fv3config                                                                                                                                 

In [2]: config =fv3config.get_default_config()                                                                                                           

In [3]: fv3config.run(config, 'test_experiment', docker_image='us.gcr.io/vcm-ml/fv3gfs-python')                                                          
Traceback (most recent call last):
  File "/usr/lib/python3.7/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/lib/python3.7/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/fv3gfs-python/external/fv3config/fv3config/fv3run/__main__.py", line 35, in <module>
    sys.exit(main())
  File "/fv3gfs-python/external/fv3config/fv3config/fv3run/__main__.py", line 31, in main
    run(args.config, args.outdir, args.runfile, args.dockerimage, args.keyfile)
  File "/fv3gfs-python/external/fv3config/fv3config/fv3run/_run.py", line 15, in run
    _run_native(config_dict_or_location, outdir, runfile=runfile)
  File "/fv3gfs-python/external/fv3config/fv3config/fv3run/_native.py", line 29, in _run_native
    localdir, n_processes, runfile_name=_get_basename_or_none(runfile))
  File "/usr/lib/python3.7/contextlib.py", line 119, in __exit__
    next(self.gen)
  File "/fv3gfs-python/external/fv3config/fv3config/fv3run/_native.py", line 51, in _temporary_directory
    _copy_directory(tempdir, outdir)
  File "/fv3gfs-python/external/fv3config/fv3config/fv3run/_gcloud.py", line 20, in _copy_directory
    _copy_file(source, dest, fs)
  File "/fv3gfs-python/external/fv3config/fv3config/fv3run/_gcloud.py", line 29, in _copy_file
    shutil.copy2(source_filename, dest_filename, follow_symlinks=True)
  File "/usr/lib/python3.7/shutil.py", line 266, in copy2
    copyfile(src, dst, follow_symlinks=follow_symlinks)
  File "/usr/lib/python3.7/shutil.py", line 121, in copyfile
    with open(dst, 'wb') as fdst:
PermissionError: [Errno 13] Permission denied: '/outdir/co2historicaldata_2012.txt'
---------------------------------------------------------------------------
CalledProcessError                        Traceback (most recent call last)
<ipython-input-3-3bb6607f5ad4> in <module>
----> 1 fv3config.run(config, 'test_experiment', docker_image='us.gcr.io/vcm-ml/fv3gfs-python')

~/workspace/fv3config/fv3config/fv3run/_run.py in run(config_dict_or_location, outdir, runfile, docker_image, keyfile)
     27         _run_in_docker(
     28             config_dict_or_location, outdir, docker_image,
---> 29             runfile=runfile, keyfile=keyfile
     30         )
     31     else:

~/workspace/fv3config/fv3config/fv3run/_docker.py in _run_in_docker(config_dict_or_location, outdir, dockerimage, runfile, keyfile)
     28         subprocess.check_call(
     29             DOCKER_COMMAND + bind_mount_args + docker_args + [dockerimage] +
---> 30             python_command + python_args)
     31 
     32 

~/miniconda3/envs/fv3net/lib/python3.7/subprocess.py in check_call(*popenargs, **kwargs)
    345         if cmd is None:
    346             cmd = popenargs[0]
--> 347         raise CalledProcessError(retcode, cmd)
    348     return 0
    349 

CalledProcessError: Command '['docker', 'run', '-v', '/tmp/tmpepr2szo0.yaml:/fv3config.yaml', '-v', '/home/noahb/workspace/fv3config/examples/test_experiment:/outdir', '--user', '1000:1000', 'us.gcr.io/vcm-ml/fv3gfs-python', 'python3', '-m', 'fv3config.fv3run', '/fv3config.yaml', '/outdir']' returned non-zero exit status 1.
```